src/geodesy.jl

```diff
@@ -0,0 +1,447 @@
+end
+
+import Base.isless
+import Base.isgreater
+import DataStructures.Queue, DataStructures.Deque, DataStructures.enqueue!, DataStructures.dequeue!, DataStructures.dequeue_pair!
+
+
+"""
+    hmaxima(image, connectivity=[], constant) -> Array
+The h-maxima transformation suppresses all maxima whose "relative" depth is below a given threshold (constant)
+# Arguments
+- `image::AbstractArray{T, N}`: where {T<:ImageCore.NumberLike , N} the ``N``-dimensional input image
+- `connectivity::AbstractArray{Bool}`: the neighborhood connectivity.
+- `constant::T`: the soustracted constant.
+# Returns
+An array of the same type and shape as the `image`.
+# See also
+[`hminima`](@ref)
+# References
+    Morphological image analysis by Soille pg 170-172
+"""
+function hmaxima(image::AbstractArray{T, N}, connectivity::AbstractArray{Bool}, constant::T) where {T<:ImageCore.NumberLike , N}
+    tmp = image.-constant
+    return underbuild(tmp,image,connectivity) 
+end
+
+"""
+    hminima(image, connectivity=[], constant) -> Array
+The h-minima transformation suppresses all minima whose "relative" depth is below a given threshold (constant)
+# Arguments
+- `image::AbstractArray{T, N}`: where {T<:ImageCore.NumberLike , N} the ``N``-dimensional input image
+- `connectivity::AbstractArray{Bool}`: the neighborhood connectivity.
+- `constant::T`: the added constant.
+# Returns
+An array of the same type and shape as the `image`.
+# See also
+[`hmaxima`](@ref)
+# References
+    Morphological image analysis by Soille pg 170-172
+"""
+function hminima(image::AbstractArray{T, N}, connectivity::AbstractArray{Bool}, constant::T) where {T<:ImageCore.NumberLike , N}
+    tmp = image.+constant
+    return overbuild(tmp,image,connectivity)    
+end
+
+"""
+    underbuild(marker, mask, connectivity=[]) -> Array
+
+Morphological reconstruction by dilatation according to connection type
+
+# Details
+This operation performs dilations of marker restricted by the provided mask until idempotence This means that the output image
+will be included between mask and min(marker,mask).
+# Arguments
+- `marker::AbstractArray{T, N}`: where {T<:ImageCore.NumberLike , N} the ``N``-dimensional input image
+- `mask::AbstractArray{T, N}`: where {T<:ImageCore.NumberLike , N} the ``N``-dimensional input image
+- `connectivity::AbstractArray{Bool}`: the neighborhood connectivity.
+# Returns
+An array of the same type and shape as the `marker`.
+# See also
+[`overbuild`](@ref)
+# References
+  Morphological Grayscale Reconstruction in Image Analysis: Applications and Efficient Algorithms IEEE Trans Image Process. 1993;2(2)
+"""
+function underbuild(marker::AbstractArray{T, N}, mask::AbstractArray{T, N}, connectivity::AbstractArray{Bool}) where {T<:ImageCore.NumberLike , N}
+    check_image_consistency(mask, marker)
+    # WARNING
+    #  The original algorithm assume that the marker < mask, this assertion is not always true
+    #  so instead we extract the start from min(marker,mask)
```

Comment:
Just a confirmation, this comment belongs to line `output = map(min, marker, mask)` right?

Reply:
Yes i will move the comment